server/channels/api4/system.go

```diff
@@ -642,13 +660,20 @@ func getRedirectLocation(c *Context, w http.ResponseWriter, r *http.Request) {
 	res, err := client.Head(url)
 	if err != nil {
 		// Cache failures to prevent retries.
-		redirectLocationDataCache.SetWithExpiry(url, "", RedirectLocationCacheExpiry)
+		cacheErr := redirectLocationDataCache.SetWithExpiry(url, "", RedirectLocationCacheExpiry)
+		if cacheErr != nil {
+			c.Logger.Warn("failed to set cache for url", mlog.Err(err))
```

Comment:
Let's log the URL as well
```suggestion
			c.Logger.Warn("failed to set cache for URL", mlog.String("url", url), mlog.Err(err))
```

Reply:
Done!